src/main/java/org/folio/linked/data/service/resource/meta/MetadataServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package org.folio.linked.data.service.resource.meta;
+
+import static java.util.Objects.nonNull;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.INSTANCE;
+import static org.folio.linked.data.model.entity.ResourceSource.LINKED_DATA;
+
+import java.util.UUID;
+import org.folio.linked.data.model.entity.InstanceMetadata;
+import org.folio.linked.data.model.entity.Resource;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MetadataServiceImpl implements MetadataService {
```

Comment:
minor
I really don't like working with null, passing null, and dancing with null check
That's why my suggestion is ...
```
  @Override
  public void ensureMetadata(Resource resource) {
    ensure(resource, metadata -> {
      metadata.setInventoryId(UUID.randomUUID().toString());
    });
  }

  @Override
  public void ensureMetadata(Resource resource, InstanceMetadata oldResourceMeta) {
    ensure(resource, metadata -> {
      metadata.setInventoryId(UUID.randomUUID().toString());
      metadata.setSrsId(oldResourceMeta.getSrsId());
    });
  }

  private void ensure(Resource resource, Consumer<InstanceMetadata> metadataConsumer) {
    if (resource.isNotOfType(INSTANCE)) {
      return;
    }
    var metadata = new InstanceMetadata(resource)
      .setResource(resource)
      .setSource(LINKED_DATA);
    metadataConsumer.accept(metadata);
    resource.setInstanceMetadata(metadata);
  }
  ```
  and not expected `null` as parameter at all

Reply:
I like this, thank you! Will think about it next time :)